python/cudf/cudf/utils/dtypes.py

```diff
@@ -578,6 +583,29 @@ def _get_nan_for_dtype(dtype):
         return np.float64("nan")
 
 
+def _decimal_to_int64(decimal: Decimal) -> int:
+    """
+    Convert a decimal.Decimal value to the int64
+    expected by libcudf
+    """
+    if int(decimal.radix()) != 10:
+        raise ValueError("Only base-10 decimals currently supported.")
+    digits = decimal.as_tuple().digits
+    start = 0
+    for power, digit in enumerate(reversed(digits)):
+        start += digit * (10 ** power)
+    return start
```

Comment:
This won't handle the case where there's an exponent: `decimal.Decimal("1.23E10")`. Regardless, I actually think it's faster to use string processing in this situation anyway and f-strings make this pretty easy:
```
def _decimal_to_int64(decimal: Decimal) -> int:
    return int(f"{decimal:0f}".replace(".", ""))
```

Reply:
I think this method will work for exponents because `decimal.Decimal` parses it nicely. Though string method route is certainly faster. Benchmark code:

```python
import timeit
start = """
import functools
import decimal
import random
x = decimal.Decimal(random.randint(0, 10**8))
"""
code1 = """
y = functools.reduce(lambda acc, d: acc*10+d, x.as_tuple().digits)
"""

code2 = """
y = int(f"{x:0f}".replace(".", ""))
"""

# [2.0489314938895404, 2.0028218089137226, 2.007592200068757, 1.9889525739708915, 1.9985147500410676]
print(timeit.repeat(stmt=code1, setup=start))
# [0.568572816089727, 0.5747550230007619, 0.5749346490483731, 0.5729447479825467, 0.5851611510151997]
print(timeit.repeat(stmt=code2, setup=start))
```

Reply:
I just have one small concern that `f"{decimal:0f}".replace(".", "")` will not implicitly convert the decimal into float representation and loses accuracy.

Reply:
I tried @kkraus14 function for a bunch of examples and wasn't able to find a case where the float conversion caused an incorrect result to be produced. There's certainly some precision issues constructing a decimal from a float, but I don't encounter the same issue converting a decimal to a string through these means. It seems to work based off python's implicit rounding in the float class `__repr__`, which works really well. 

For completeness, I found that the current function can be adjusted to work for the exponent cases by changing the iteration to 

```
start += digit * (10 ** (power + exponent))
``` 

But let's roll with the string approach for now.

Reply:
I think it's more than that, I tried it with arbitrarily large floating point numbers versus the "equivalent" decimal and the results for strings are different:
```
import decimal

a = 1.23E100
b = decimal.Decimal("1.23E100")

print(f"{a:0f}")
# 12300000000000000836686295084537585379506223785413935301425289783235883702867663918638982200322686976.000000
print(f"{b:0f}")
# 12300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
```

Reply:
right, that's because `a` is an inexact float and the `Decimal` constructor gives back a `Decimal` which is equal to exactly what the float really is. 

The purpose of `_decimal_to_int64` is to get the right int though and I have not seen an example where it gives the wrong int for a `Decimal` that is properly constructed (speaking about the string version).

Reply:
Looking at `__format__` class of decimal, it seems like the only case where the the format may be imprecise is when the specified precision fewer than the decimal precision. Otherwise it's only operating in string space to maintain accuracy.